### PR TITLE
chore(ui-react): add jest coverage and thresholds

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add jest coverage thresholds to _@aws-amplify/ui-react_
- threshold values are rounded down to the nearest whole number of their current value

#### Issue #, if available
NA

#### Description of how you validated changes
`yarn react test`
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/jest.config.js [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+  // ignore coverage for top level "export", PrimitiveCatalog, and style files
+  coveragePathIgnorePatterns: [
+    '<rootDir>/src/(index|internal|legacy|PrimitiveCatalog|styles).(ts|tsx)',
+  ],
+  coverageThreshold: {
+    global: {
+      branches: 71,
+      functions: 60,
+      lines: 79,
+      statements: 79,
```
Comment (thaddmt) [inline packages/react/jest.config.js:13]: I'm guessing this is just where our coverage is currently at?

**Review by thaddmt — APPROVED** [review]
LGTM

File: packages/react/jest.config.js (hunk 1 above)
Comment (calebpollman) [reply packages/react/jest.config.js:13]: That's correct 😄

Comment (0618) [reply packages/react/jest.config.js:13]: Our current coverage is surprisingly higher than I thought 😂

**Review by 0618 — APPROVED** [review]
Thanks for raising the testing bar!

**Review by ErikCH — APPROVED** [review]
As we continue on, I imagine we'll find us in scenarios where some code is not testable, and if so, we can ignore that code from our test coverage. 

Otherwise, we are good to go.

**Review by wlee221 — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**